src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -367,12 +367,66 @@ def update_webapp(instance, client_affinity_enabled=None, https_only=None):
     return instance
 
 
+def update_functionapp(cmd, instance, plan=None):
+    client = web_client_factory(cmd.cli_ctx)
+    if plan is not None:
+        if is_valid_resource_id(plan):
+            dest_parse_result = parse_resource_id(plan)
+            dest_plan_info = client.app_service_plans.get(dest_parse_result['resource_group'],
+                                                          dest_parse_result['name'])
+        else:
+            dest_plan_info = client.app_service_plans.get(instance.resource_group, plan)
+        if dest_plan_info is None:
+            raise CLIError("The plan '{}' doesn't exist".format(plan))
+        validate_plan_switch_compatibility(client, instance, dest_plan_info)
+        instance.server_farm_id = dest_plan_info.id
+    return instance
+
+
+def validate_plan_switch_compatibility(client, src_functionapp_instance, dest_plan_instance):
+    general_switch_msg = 'Currently the switch is only allowed between a Consumption or an Elastic Premium plan.'
+    src_parse_result = parse_resource_id(src_functionapp_instance.server_farm_id)
+    src_plan_info = client.app_service_plans.get(src_parse_result['resource_group'],
+                                                 src_parse_result['name'])
+    if src_plan_info is None:
+        # if kind == functionapp, it means the app is using a Consumption plan
+        # Sometimes retrieving the exact plan fails, so it's ok as long as we know they are in Consumption
+        if src_functionapp_instance.kind.lower() != 'functionapp':
+            raise CLIError('Could not determine the current plan of the functionapp')
+    elif not (is_plan_consumption(src_plan_info) or is_plan_Elastic_Premium(src_plan_info)):
+        raise CLIError('Your functionapp is not using a Consumption or an Elastic Premium plan. ' + general_switch_msg)
+    if not (is_plan_consumption(dest_plan_instance) or is_plan_Elastic_Premium(dest_plan_instance)):
+        raise CLIError('You are trying to move to a plan that is not a Consumption or an Elastic Premium plan. ' +
+                       general_switch_msg)
+
+
 def set_functionapp(cmd, resource_group_name, name, **kwargs):
     instance = kwargs['parameters']
     if 'function' not in instance.kind:
         raise CLIError('Not a function app to update')
     client = web_client_factory(cmd.cli_ctx)
-    return client.web_apps.create_or_update(resource_group_name, name, site_envelope=instance)
+    try:
+        return client.web_apps.create_or_update(resource_group_name, name, site_envelope=instance)
+    except DefaultErrorResponseException as ex:
```

Comment:
@ankitkumarr for exception handling of this sort we have a helper in commands.py we initiate it by adding this to the command on the commands.py exception_handler=ex_handler_factory(), ex. see   g.custom_command('create', 'create_function', exception_handler=ex_handler_factory()) -> can you check if this is something you can utilize here for this method as well?

Reply:
Thanks @panchagnula, I will update it to use that factor directly.